2a. React/1. Basics/solvro-projects-combobox.test.tsx

```diff
@@ -0,0 +1,122 @@
+import { describe, expect, it, vi } from "vitest";
+import { render, screen, within } from "@testing-library/react";
+import { SolvroProjectsCombobox } from "../src/components/solvro-projects-combobox";
+import "@testing-library/jest-dom/vitest";
+import userEvent from "@testing-library/user-event";
+import { beforeEach } from "node:test";
+
+// needed for Radix UI components
+class ResizeObserver {
+  observe = vi.fn();
+  unobserve = vi.fn();
+  disconnect = vi.fn();
+}
+window.ResizeObserver = ResizeObserver;
+
+Element.prototype.scrollIntoView = vi.fn();
+
+describe("SolvroProjectsCombobox", () => {
+  beforeEach(() => {
+    vi.clearAllMocks();
+  });
+
+  const setup = () => {
+    const user = userEvent.setup();
+    render(<SolvroProjectsCombobox />);
+
+    const getComboboxButton = () => screen.getByRole("combobox");
+    const getPopover = () => document.querySelector('[data-state="open"]');
+    const getOptions = () => screen.queryAllByRole("option");
+    const getSearchInput = () => screen.getByPlaceholderText(/projekt/i);
+    const getEmptyMessage = () => screen.queryByText(/znaleziono projekt/i);
+
+    const openCombobox = async () => {
+      await user.click(getComboboxButton());
+    };
+
+    const searchFor = async (text: string) => {
+      const input = getSearchInput();
+      await user.clear(input);
+      await user.type(input, text);
+      return getOptions();
+    };
+
+    const selectOption = async (index: number) => {
+      const options = getOptions();
+      if (options[index]) {
+        await user.click(options[index]);
+      }
+    };
+
+    return {
+      user,
+      getComboboxButton,
+      getPopover,
+      getOptions,
+      getSearchInput,
+      getEmptyMessage,
+      openCombobox,
+      searchFor,
+      selectOption,
+    };
+  };
+
+  it("should render", () => {
+    const { getComboboxButton } = setup();
+    expect(getComboboxButton()).toBeInTheDocument();
+  });
+
+  it("should open popover when clicked", async () => {
+    const { openCombobox, getOptions } = setup();
+    await openCombobox();
+    expect(getOptions()).toHaveLength(5);
+  });
+
+  it("should filter options when searching", async () => {
+    const { openCombobox, searchFor } = setup();
+    await openCombobox();
+
+    const options = await searchFor("topwr");
```

Comment:
it feels a bit weird to see the variable named `options` when you have a `getOptions` function and you don't use it here